src/etl/ETLSource.h

```diff
@@ -323,6 +341,7 @@ class ETLSourceImpl : public ETLSource
                              << " . Remote = " << toString();
             }
         }
+        tokenStr_ = fmt::format("{}:{}:{}", ip_, wsPort_, grpcPort_);
```

Comment:
I'm not sure we can do it this way: i think it's a legitimate usecase to have more than one of the same ETL servers configured in the config so that they just participate in load balancing on clio side. I suppose in this situation your code will get confused. 
What i think we can do is implement `operator==` properly and compare `this` pointer inside. Comparing two Probing sources would just use the underlying source's `operator==`.
Comparing by `this` is not that bad if it's hidden inside. Comparing by `shared_ptr.get()` on the other hand is really bad.

Reply:
Even multiple ETL servers configured in the config, they can be distinguished by ip+port. 
Here the comparison is not among two probe sources, it is between one probe and one underlying source. Let us discuss offline.